### PR TITLE
Security Considerations

### LINKED ISSUE #19: Address Missing Security Considerations
The document lacks a Security Considerations section. It may be that there are no new security considerations beyond those in 4329, but if that is the case this document needs to explicitly say (and justify) that.  But please consider whether it is realistic to say we’ve learned nothing new about the security considerations since 4329 was published 13 years ago. (We can expect statements to that effect to get close scrutiny from the secdir reviewer and security ADs.)

### PR DESCRIPTION
**Pull request by linuxwolf**
Add a Security Considerations, updating the original considerations in RFC 4329:

* Modules and extra scripts
* SharedArrayBuffer

Fixes #19

### CODE REVIEW

**Comment by linuxwolf** [comment]
CC @annevk and @lukewagner

File: javascript-mjs/draft.md [hunk 1]
```diff
@@ -82,6 +109,14 @@ This separation means that (in the absence of additional information) there are
...
 
+# Security Considerations
+
+Module scripts in ECMAScript can request the fetching and processing of additional scripts.  Implementations that support modules need to ensure these scripts are treated the same as scripts scripts directly processed.
+
+ECMAScript prior to version 8 was not considered to support dangerous operations.  However, the addition of SharedArrayBuffers in ECMAScript version 8 enable access to shared memory blocks between host environments, which can allow for input and output of data with external sources.  This has lead to certain types of timing attacks {{SPECTRE}}.  Implementations may wish to take steps to mitigate this concern, such as disabling or removing support for SharedArrayBuffers, or take additional steps to ensure access to this shared memory is only accessible between execution contexts that have some form of mutual trust.
```
Comment (annevk) [inline javascript-mjs/draft.md]: > which can allow for input and output of data with external sources

Is that true? I think this should say that they allow for the creation of a high-resolution timer.

File: javascript-mjs/draft.md [hunk 2]
```diff
@@ -82,6 +109,14 @@ This separation means that (in the absence of additional information) there are
 
 It is not possible to fully determine if a Source Text of ECMAScript is meant to be parsed in the Module or Script grammar goals based upon content alone. Therefore, scripting environments must use out of band information in order to determine what goal a Source Text should be treated as. To this end some scripting environments have chosen to adopt a new file extension of .mjs for determining the goal of a given Source Text.
 
+# Security Considerations
+
+Module scripts in ECMAScript can request the fetching and processing of additional scripts.  Implementations that support modules need to ensure these scripts are treated the same as scripts scripts directly processed.
```
Comment (annevk) [inline javascript-mjs/draft.md]: You might want to mention that depending on the host there could be security or privacy implications as a result of this fetching.

File: javascript-mjs/draft.md (hunk 1 above)
Comment (linuxwolf) [reply javascript-mjs/draft.md]: That's fair; I was drawing conclusions from reading SPECTRE again, but I can't really back it up.  I'll reword to only talk about high-resolution timers.

Comment (annevk) [reply javascript-mjs/draft.md]: Are you thinking of https://en.wikipedia.org/wiki/Row_hammer perhaps? There are some risks there I suppose depending on the hardware.

**Pull request merged**